nemo_curator/modules/semantic_dedup/semdedup.py

```diff
@@ -90,19 +92,47 @@ def __init__(
         )
         self.eps_to_extract = config.eps_to_extract
 
-    def call(self, dataset: DocumentDataset) -> DocumentDataset:
+    def identify_duplicates(self, dataset: DocumentDataset) -> DocumentDataset:
         """
-        Execute the SemDedup process.
-
-        Args:
-            dataset (DocumentDataset): Input dataset for deduplication.
-
-        Returns:
-            DocumentDataset: Deduplicated dataset.
+        Identify duplicates in the dataset. Returns a list of ids that are duplicates to each other.
         """
         embeddings_dataset = self.embedding_creator(dataset)
         self.clustering_model(embeddings_dataset)
         self.semantic_cluster_dedup.compute_semantic_match_dfs()
         return self.semantic_cluster_dedup.extract_dedup_data(
             eps_to_extract=self.eps_to_extract
         )
+
+    def remove(
+        self, dataset: DocumentDataset, duplicates_to_remove: DocumentDataset
+    ) -> DocumentDataset:
+        """
+        Remove duplicates from the dataset.
+        """
+        result = remove_duplicates(
+            dataset.df,
+            duplicates_to_remove.df,
+            self.id_column,
+            group_field=None,
+            perform_shuffle=False,
+        )
+        return DocumentDataset(result)
+
+    def call(
+        self, dataset: DocumentDataset, perform_removal: bool = False
+    ) -> DocumentDataset:
```

Comment:
Prefer not to have `perform_removal` in the call function.

Reply:
That's what we also have in FuzzyDedup, happy to change it, any suggestions?

Reply:
Yes, I thought the consensus was to have it in the config instead of call - https://github.com/NVIDIA/NeMo-Curator/pull/509/files#r1941748261 ? Looks like that did not end up happening for fuzzy deduplication. Was there a reason why?

Reply:
Looks like I might've missed pushing the commit 😨 
Let me fix it in this. do you think those should live in SemDedupConfig / FuzzyDedupConfig or in init? I think init, given the configs are already soooo bloated